src/main/javascript/controller/cat-base-detail-controller.js

```diff
@@ -157,18 +157,29 @@ function CatBaseDetailController($scope, $state, $stateParams, $location, $windo
      * Upon success the view mode of the details of the currently created / updated object will be shown.
      * Upon an error the reported errors (global & field errors) will be shown to the user and the edit mode
      * will remain active.
+     *
+     * * @param {object} stayInEdit If true the view stays in detail edit state after save instead of switching to
+     *                              detail view state.
      */
-    $scope.save = function () {
-        endpoint.save($scope.editDetail).then(function (data) {
+    $scope.save = function (stayInEdit) {
+        // When passing data to an asynchronous method, it makes sense to clone it.
+        endpoint.save(angular.copy($scope.editDetail)).then(function (data) {
```

Comment:
Hi Thomas,

no, I actually had the problem.
Wenn du das folgende tust:
var myDto = {….};
// Dann wird hier unter Umständen auch der Wert 10 bereits gespeichert, da save asynchron auf myDto zugreift, es ist aber klarer, wenn wirklich der Zustand gespeichert wird, der an save übergeben wird..

endpoint.save(myDto);

myDto.anyValue = 10;





Lg

Michael



Von: Thomas Scheinecker [mailto:notifications@github.com]
Gesendet: Montag, 2. März 2015 09:37
An: Catalysts/cat-angular
Cc: KURZ Michael
Betreff: Re: [cat-angular] -added: optional parameter "stayInEdit" to save() method, so that the co... (#4)


In src/main/javascript/controller/cat-base-detail-controller.js<https://github.com/Catalysts/cat-angular/pull/4#discussion_r25583716>:

>       */

> -    $scope.save = function () {

> -        endpoint.save($scope.editDetail).then(function (data) {

> +    $scope.save = function (stayInEdit) {

> +        // When passing data to an asynchronous method, it makes sense to clone it.

> +        endpoint.save(angular.copy($scope.editDetail)).then(function (data) {

// When passing data to an asynchronous method, it makes sense to clone it.
where does this come from?
this is completely unnecessary

—
Reply to this email directly or view it on GitHub<https://github.com/Catalysts/cat-angular/pull/4/files#r25583716>.